Supervision Work/src/uk/ac/cam/ap886/oopjava/supervisionProjects/Hangman/ArrayDictionary.java

```diff
@@ -0,0 +1,50 @@
+package uk.ac.cam.ap886.oopjava.supervisionProjects.Hangman;
+
+import java.io.BufferedReader;
+import java.io.FileNotFoundException;
+import java.io.FileReader;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Random;
+
+public class ArrayDictionary implements Dictionary {
+	private ArrayList <String> words=new ArrayList<String>();
+
+	//------------------------------------------
+	public ArrayDictionary(String filename){
+		try (BufferedReader r = new BufferedReader (new FileReader (filename));){
+			String temp = "";
+			do{
+				temp= r.readLine();
+				words.add(temp);
+			}while(temp!=null);
+		} catch (FileNotFoundException e) {
+			e.printStackTrace();	//Shouldn't happen
+		} catch (IOException e) {
+			e.printStackTrace();	//In case of Empty file
+		}
+		assert(!words.isEmpty());
```

Comment:
I don't see what's wrong with an empty datastructure. Again, a dictionary is just a datastructure, it implies nothing about what its being used for. The caller should be responsible for checking that the dictionary is properly populated.